RFS/src/main/java/com/rfs/common/FilterScheme.java

```diff
@@ -0,0 +1,23 @@
+package com.rfs.common;
+
+import java.util.List;
+import java.util.function.BiConsumer;
+import java.util.function.Predicate;
+
+public class FilterScheme {
+
+    public static Predicate<SnapshotRepo.Index> filterIndicesByAllowList(List<String> indexAllowlist, BiConsumer<String, Boolean> indexNameAcceptanceObserver) {
+        return index -> {
+            boolean accepted;
+            if (indexAllowlist.isEmpty()) {
+                accepted = !index.getName().startsWith(".");
+            } else {
+                accepted = indexAllowlist.contains(index.getName());
```

Comment:
Let's make a backlog item for this to support regex

Reply:
+1

Reply:
Good idea, done: https://opensearch.atlassian.net/browse/MIGRATIONS-1804